runtime/lua/vim/lsp/protocol.lua

```diff
@@ -312,11 +310,23 @@ local constants = {
   },
 }
 
-for k, v in pairs(constants) do
-  local tbl = vim.deepcopy(v)
-  vim.tbl_add_reverse_lookup(tbl)
-  protocol[k] = tbl
-end
+local protocol = vim.deepcopy(constants)
+
+--- @nodoc
+--- @type table<lsp.CompletionItemKind,string>
+protocol.CompletionItemKindInv = vim.tbl_inv(protocol.CompletionItemKind)
```

Comment:
Could extend the existing tables instead of creating separate "Inv" variants. That's an ok pattern for fixed collections where we know the values won't have conflicts. I just don't think it needs to be baked into a `tbl_add_reverse_lookup` function.